### DISCUSSION TITLE
Bean overriding in case of multiple beans for the same class are present.

### DISCUSSION THREAD

**Discussion by ironhide0396** [gruelbox/transaction-outbox#681] [Ideas]
It is quite a normal use case for developers to have a baseBean in the production files and have an overridden one in the test files. The bean in the test file is marked with `@Primary` so that it takes precedence when being invoked from tests.

Now, in case of using `transactionoutbox-spring`, the following method 
```
@Override
public String getName(Class<?> clazz) {
    String[] beanNames = applicationContext.getBeanNamesForType(clazz);
    if (beanNames.length > 1) {
      throw new IllegalArgumentException(
          "Type "
              + clazz.getName()
              + " exists under multiple names in the context: "
              + Arrays.toString(beanNames)
              + ". Use a unique type.");
    }
    if (beanNames.length == 0) {
      throw new IllegalArgumentException(
          "Type " + clazz.getName() + " not available in the context");
    }
    return beanNames[0];
}
``` 
throws errors when there are more than 2 beans for the same class. Should we be adding support to check if any of the beans have precedence set via `@Primary` annotation and maybe select that ?

**Comment by badgerwithagun** [comment]
Seems to make sense, yes! 👍🏻 

Will you be submitting a PR?
